### PR TITLE
Use raw docstrings instead of escaping backslashes

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Note: I would like to enforce [pydocstyle D301](http://www.pydocstyle.org/en/stable/error_codes.html) on this, but we have several cases of backslash as line continuation which would give false-positives. For now, we cannot check this automatically.

### CODE REVIEW

File: lib/matplotlib/ticker.py [hunk 1]
```diff
@@ -1296,7 +1296,7 @@ def __call__(self, x, pos=None):
         return self.fix_minus(s)
 
     def format_eng(self, num):
-        """
+        r"""
```
Comment (QuLogic) [inline lib/matplotlib/ticker.py]: I don't think this is right; the escape is in an output, so the interpreted string is correct.

File: lib/matplotlib/colors.py [hunk 2]
```diff
@@ -1986,8 +1986,7 @@ def from_levels_and_colors(levels, colors, extend='neither'):
 
     Returns
     -------
-    (cmap, norm) : tuple containing a :class:`Colormap` and a \
-                   :class:`Normalize` instance
+    cmap, norm : `.Colormap`, `.Normalize`
```
Comment (QuLogic) [inline lib/matplotlib/colors.py]: Multiple returns should be [separate entries](https://github.com/numpy/numpydoc/blob/master/doc/example.py#L57).

File: lib/matplotlib/ticker.py (hunk 1 above)
Comment (timhoffm) [reply lib/matplotlib/ticker.py]: Thanks, you are right!

**Comment by QuLogic** [comment]
> but we have several cases of backslash as line continuation which would give false-positives.

That seems like a bug in pydocstyle that should be reported to them.

**Comment by timhoffm** [comment]
Already done: PyCQA/pydocstyle#364

**Pull request merged**